### PR TITLE
Add ifort on Windows

### PR DESCRIPTION
**Pull request by scivision**
After #5331 , Intel Fortran compiler `ifort` on Windows works and passes the Fortran tests from run_project_tests.py.

Implementing this was mainly a matter of 

* adding one [forgotten argument](https://github.com/mesonbuild/meson/compare/master...scivision:icl_ifort?expand=1#diff-64440ffbb1d8b859bdc6286f6a9c38c0R274)
* making sure the ifort compiler made .obj files in the test [build directory](https://github.com/mesonbuild/meson/compare/master...scivision:icl_ifort?expand=1#diff-64440ffbb1d8b859bdc6286f6a9c38c0R72)
* straighten out [logic for detecting Ninja for ifort on Windows](https://github.com/mesonbuild/meson/compare/master...scivision:icl_ifort?expand=1#diff-d9f7c80135592b1a61e38f3133a89499R565)
* type hinting
* fix typo for PGI, just happened to see that.

### CODE REVIEW

**Comment by dcbaker** [comment]
Thanks for working on this, it all looks good, but I'll give it a double check on my windows machine before I give my rb

**Comment by jpakkane** [comment]
:crossed_swords: conflicts. But other than that LGTM, feel free to merge if ok by you.

**Comment by scivision** [comment]
Corrected merge conflicts. 

@dcbaker improvements on `master` #5515  rebased here: you can check `ifort` on your windows machine in a minute or two with
```sh
python run_project_tests.py --only fortran
```

File: mesonbuild/dependencies/misc.py [hunk 1]
```diff
@@ -213,7 +217,7 @@ def __init__(self, environment, kwargs):
 
         if not self.is_found and mesonlib.is_windows():
             # only Intel Fortran compiler is compatible with Microsoft MPI at this time.
-            if language == 'fortran' and environment.detect_fortran_compiler(self.for_machine).name_string() != 'intel':
+            if language == 'fortran' and environment.detect_fortran_compiler(self.for_machine).name_string() != 'intel-ci':
```
Comment (dcbaker) [inline mesonbuild/dependencies/misc.py]: that should be `intel-cl` not `intel-ci`, right?

Comment (scivision) [reply mesonbuild/dependencies/misc.py]: Thanks! corrected in f990c3e

**Comment by dcbaker** [comment]
Once the CI comes back I'll merge this.

**Pull request merged**